http/src/main/java/org/sonarsource/sonarlint/core/http/HttpClientProvider.java

```diff
@@ -61,25 +67,47 @@ public HttpClientProvider(String userAgent, @Nullable Path sonarlintUserHome,
       var truststorePath = System.getProperty("sonarlint.ssl.trustStorePath", requireNonNull(sonarlintUserHome).resolve("ssl/truststore.p12").toString());
       sslFactoryBuilder.withInflatableTrustMaterial(Paths.get(truststorePath), TRUSTSTORE_PWD, "PKCS12", trustManagerParametersPredicate);
     }
+    var connectionConfigBuilder = ConnectionConfig.custom();
+    getTimeoutFromSystemProp("sonarlint.http.connectTimeout").ifPresent(connectionConfigBuilder::setConnectTimeout);
+    getTimeoutFromSystemProp("sonarlint.http.socketTimeout").ifPresent(connectionConfigBuilder::setSocketTimeout);
     var asyncConnectionManager = PoolingAsyncClientConnectionManagerBuilder.create()
       .setTlsStrategy(new DefaultClientTlsStrategy(sslFactoryBuilder.build().getSslContext()))
       .setDefaultTlsConfig(TlsConfig.custom()
         // Force HTTP/1 since we know SQ/SC don't support HTTP/2 ATM
         .setVersionPolicy(HttpVersionPolicy.FORCE_HTTP_1)
         .build())
+      .setDefaultConnectionConfig(connectionConfigBuilder.build())
       .build();
+    var requestConfigBuilder = RequestConfig.custom();
+    getTimeoutFromSystemProp("sonarlint.http.connectionRequestTimeout").ifPresent(requestConfigBuilder::setConnectionRequestTimeout);
+    getTimeoutFromSystemProp("sonarlint.http.responseTimeout").ifPresent(requestConfigBuilder::setResponseTimeout);
     this.sharedClient = HttpAsyncClients.custom()
       .setConnectionManager(asyncConnectionManager)
       .addResponseInterceptorFirst(new RedirectInterceptor())
       .setUserAgent(userAgent)
       // proxy settings
       .setRoutePlanner(new SystemDefaultRoutePlanner(proxySelector))
       .setDefaultCredentialsProvider(proxyCredentialsProvider)
+      .setDefaultRequestConfig(
+        requestConfigBuilder
+          .build())
       .build();
 
     sharedClient.start();
   }
 
+  private static Optional<Timeout> getTimeoutFromSystemProp(String key) {
+    return Optional.ofNullable(System.getProperty(key))
+      .map(s -> {
+        try {
+          return Duration.ofMinutes(Integer.parseInt(s));
```

Comment:
Does not look very clear to me what's expected. Is this supposed to be documented or simply explained on case-by-case troubleshooting?

So you can provide either a number for minutes or a string of pattern `PnDTnHnMn.nS` for the duration parsing?

Reply:
Yeah, for most users it will be minutes, but for tests, I needed to allow smaller durations. I considered to only accept `PnDTnHnMn.nS` but I think this is not very user friendly. WDYT?

Reply:
I'm not strongly opinionated on this one, personally, I'd have gone for seconds only. In case it might be beneficial to have smaller granularity (milliseconds), then why not keep it as it is now.